engine/api/workflow/resync_workflow.go

```diff
@@ -38,6 +39,7 @@ func Resync(db gorp.SqlExecutor, store cache.Store, proj *sdk.Project, wr *sdk.W
 	wr.Workflow.Applications = wf.Applications
 	wr.Workflow.Environments = wf.Environments
 	wr.Workflow.ProjectIntegrations = wf.ProjectIntegrations
+	wr.Workflow.EventIntegrations = wf.EventIntegrations
```

Comment:
Maybe check EventIntegrations in IsValid function like appID, envID, pipID, etc..

Reply:
done